Flake: Agtw e2e TestTimeout Flakes

```sh
=== RUN   TestTimeouts
2026/01/26 19:13:24.616394 KUBECTL: kubectl --context kind-i2g-e2e apply -f /home/runner/work/ingress2gateway/ingress2gateway/test/e2e/emitters/agentgateway/testdata/input/timeouts.yaml
2026/01/26 19:13:24.944842 KUBECTL: kubectl --context kind-i2g-e2e -n ingress-nginx get service ingress-nginx-controller -o json
2026/01/26 19:13:25.234865 KUBECTL: kubectl --context kind-i2g-e2e apply -f /tmp/i2g-generated-2609742372.yaml
2026/01/26 19:13:25.472016 KUBECTL: kubectl --context kind-i2g-e2e -n default get gateway nginx -o json
2026/01/26 19:13:25.637604 KUBECTL: kubectl --context kind-i2g-e2e -n default get gateway nginx -o json
2026/01/26 19:13:25.782343 KUBECTL: kubectl --context kind-i2g-e2e -n default get httproute ingress-timeout-proxy-read-timeout-proxy-read-example-org -o json
2026/01/26 19:13:25.890930 KUBECTL: kubectl --context kind-i2g-e2e -n default get httproute ingress-timeout-proxy-read-timeout-proxy-read-example-org -o json
2026/01/26 19:13:26.004522 KUBECTL: kubectl --context kind-i2g-e2e -n default get httproute ingress-timeout-proxy-send-timeout-proxy-send-example-org -o json
2026/01/26 19:13:26.109060 KUBECTL: kubectl --context kind-i2g-e2e -n default get httproute ingress-timeout-proxy-send-timeout-proxy-send-example-org -o json
2026/01/26 19:13:26.199703 KUBECTL: kubectl --context kind-i2g-e2e -n default get httproute ingress-timeout-send-and-read-timeout-send-and-read-example-org -o json
2026/01/26 19:13:26.288440 KUBECTL: kubectl --context kind-i2g-e2e -n default get httproute ingress-timeout-send-and-read-timeout-send-and-read-example-org -o json
2026/01/26 19:13:26.378265 KUBECTL: kubectl --context kind-i2g-e2e -n default get gateway nginx -o json
    e2e_test.go:249: 2026-01-26T19:13:26.473074122Z: Making GET request to host timeout-proxy-send.example.org via [http://172.18.255.204:/](http://172.18.255.204/)
    http.go:276: 2026-01-26T19:13:26.473961287Z: Response expectation failed for request: {URL: {Scheme:http Opaque: User: Host:172.18.255.204: Path:/ RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}, Host: timeout-proxy-send.example.org, Protocol: HTTP, Method: GET, Headers: map[X-Echo-Set-Header:[]], UnfollowRedirect: false, Server: , CertPem: <truncated>, KeyPem: <truncated>}  not ready yet: expected status code to be one of [200], got 404. CRes: &{404 146 HTTP/1.1 map[Content-Length:[146] Content-Type:[text/html] Date:[Mon, 26 Jan 2026 19:13:26 GMT]] <nil> []} (after 3.106µs)
    http.go:282: 2026-01-26T19:13:27.476650782Z: Request passed
    e2e_test.go:260: 2026-01-26T19:13:27.476691718Z: Making GET request to host timeout-proxy-read.example.org via http://172.18.255.205/
    http.go:276: 2026-01-26T19:13:28.51053794Z: Response expectation failed for request: {URL: {Scheme:http Opaque: User: Host:172.18.255.205 Path:/ RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}, Host: timeout-proxy-read.example.org, Protocol: HTTP, Method: GET, Headers: map[X-Echo-Set-Header:[]], UnfollowRedirect: false, Server: , CertPem: <truncated>, KeyPem: <truncated>}  not ready yet: expected status code to be one of [200], got 404. CRes: &{404 15 HTTP/1.1 map[Content-Length:[15] Content-Type:[text/plain] Date:[Mon, 26 Jan 2026 19:13:28 GMT]] <nil> []} (after 1.683µs)
    http.go:276: 2026-01-26T19:14:26.572221281Z: Response expectation failed for request: {URL: {Scheme:http Opaque: User: Host:172.18.255.205 Path:/ RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}, Host: timeout-proxy-read.example.org, Protocol: HTTP, Method: GET, Headers: map[X-Echo-Set-Header:[]], UnfollowRedirect: false, Server: , CertPem: <truncated>, KeyPem: <truncated>}  not ready yet: expected status code to be one of [200], got 404. CRes: &{404 15 HTTP/1.1 map[Content-Length:[15] Content-Type:[text/plain] Date:[Mon, 26 Jan 2026 19:14:26 GMT]] <nil> []} (after 59.09484708s)
    http.go:257: 2026-01-26T19:14:27.476907674Z: timeout while waiting after 59 attempts, 0/1 successes
2026/01/26 19:14:27.477060 KUBECTL: kubectl --context kind-i2g-e2e delete -f /tmp/i2g-generated-2609742372.yaml --ignore-not-found=true --wait=true --timeout=2m
2026/01/26 19:14:27.635006 KUBECTL: kubectl --context kind-i2g-e2e delete -f /home/runner/work/ingress2gateway/ingress2gateway/test/e2e/emitters/agentgateway/testdata/input/timeouts.yaml --ignore-not-found=true --wait=true --timeout=2m
--- FAIL: TestTimeouts (63.14s)
```

--- a/test/e2e/emitters/agentgateway/e2e_test.go
+++ b/test/e2e/emitters/agentgateway/e2e_test.go
@@ -31,6 +31,7 @@ import (
 var (
 	e2eSetupComplete bool
 	kubeContext      = common.KubeContext
+	defaultHostHeader   = common.DefaultHostHeader
 )
 
 func TestMain(m *testing.M) {
@@ -127,7 +128,7 @@ func e2eTestSetup(t *testing.T, inputFile, outputFile string) (context.Context, 
 		}
 	}
 	if hostHeader == "" {
-		hostHeader = "demo.localdev.me"
+		hostHeader = defaultHostHeader
 	}
 
 	// Verify expected output file exists for comparison.
@@ -181,10 +182,12 @@ func e2eTestSetup(t *testing.T, inputFile, outputFile string) (context.Context, 
 		t.Fatalf("gateway address: %v", err)
 	}
 
-	// Prefer HTTPRoute or TLSRoute hostnames if present.
+	// Prefer HTTPRoute or TLSRoute hostnames only when the input didn't specify any hosts.
 	host := hostHeader
-	if hr := testutils.FirstRouteHost(outObjs); hr != "" {
-		host = hr
+	if hostHeader == defaultHostHeader {
+		if hr := testutils.FirstRouteHost(outObjs); hr != "" {
+			host = hr
+		}
 	}
 
 	return ctx, gwAddr, host, hostHeader, ingressIP

--- a/test/e2e/emitters/common/config.go
+++ b/test/e2e/emitters/common/config.go
@@ -33,4 +33,6 @@ var (
 	KindClusterName = testutils.EnvOrDefault("KIND_CLUSTER_NAME", DefaultClusterName)
 	KeepCluster     = testutils.EnvOrDefault("KEEP_KIND_CLUSTER", "true") == "true"
 	KubeContext     = "kind-" + KindClusterName
+	// The host header to use for test requests. Must match in testdata input files.
+	DefaultHostHeader = "demo.localdev.me"
 )

--- a/test/e2e/emitters/kgateway/e2e_test.go
+++ b/test/e2e/emitters/kgateway/e2e_test.go
@@ -35,6 +35,7 @@ import (
 var (
 	e2eSetupComplete bool
 	kubeContext      string
+	DefaultHostHeader   = common.DefaultHostHeader
 )
 
 func TestMain(m *testing.M) {
@@ -131,7 +132,7 @@ func e2eTestSetup(t *testing.T, inputFile, outputFile string) (context.Context, 
 		}
 	}
 	if hostHeader == "" {
-		hostHeader = "demo.localdev.me"
+		hostHeader = DefaultHostHeader
 	}
 
 	// Verify expected output file exists for comparison.
@@ -185,10 +186,12 @@ func e2eTestSetup(t *testing.T, inputFile, outputFile string) (context.Context, 
 		t.Fatalf("gateway address: %v", err)
 	}
 
-	// Prefer HTTPRoute or TLSRoute hostnames if present.
+	// Prefer HTTPRoute or TLSRoute hostnames only when the input didn't specify any hosts.
 	host := hostHeader
-	if hr := testutils.FirstRouteHost(outObjs); hr != "" {
-		host = hr
+	if hostHeader == DefaultHostHeader {
+		if hr := testutils.FirstRouteHost(outObjs); hr != "" {
+			host = hr
+		}
 	}
 
 	return ctx, gwAddr, host, hostHeader, ingressIP

